sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -59,13 +59,32 @@ public class ChatClient {
             throw AzureError.client("Unable to form base URL.")
         }
 
+        // If applicationId is not provided bundle identifier will be used
+        // Instead set the default application id to be an empty string
+        var updatedOptions: AzureCommunicationChatClientOptions?
+        if options.telemetryOptions.applicationId == nil {
+            let apiVersion = AzureCommunicationChatClientOptions.ApiVersion(options.apiVersion)
+            let telemetryOptions = TelemetryOptions(
+                telemetryDisabled: options.telemetryOptions.telemetryDisabled,
+                applicationId: ""
+            )
+
+            updatedOptions = AzureCommunicationChatClientOptions(
+                apiVersion: apiVersion,
+                logger: options.logger,
+                telemetryOptions: telemetryOptions,
+                transportOptions: options.transportOptions,
+                dispatchQueue: options.dispatchQueue
+            )
+        }
+
         let communicationCredential = TokenCredentialAdapter(credential)
         let authPolicy = BearerTokenCredentialPolicy(credential: communicationCredential, scopes: [])
 
         let client = try ChatClientInternal(
             endpoint: endpointUrl,
             authPolicy: authPolicy,
-            withOptions: options
+            withOptions: updatedOptions ?? options
```

Comment:
@tjprescott I realized that UserAgentPolicy() uses separate TelemetryOptions() so this doesn't cover the case where someone provides UserAgentPolicy with telemetry options that have nil applicationId. 

We only want to change the applicationId to an empty string if it is nil, I'm not sure how we can do this for UserAgentPolicy()?

Reply:
The telemetryOptions you pass into your client contructor get passed into the UserAgentPolicy in your `ChatClientInternal` initializer (line 44).